lib/rides/models/ride.dart

```diff
@@ -147,13 +147,18 @@ class Ride extends Trip {
     await supabaseClient.from('rides').update({'status': status.index}).eq('id', id);
   }
 
+  Future<void> withdraw() async {
+    status = RideStatus.withdrawnByRider;
+    await supabaseClient.from('rides').update({'status': status.index}).eq('id', id);
+  }
+
   @override
   String toString() {
     return 'Ride{id: $id, in: $driveId, from: $start at $startTime, to: $end at $endTime, by: $riderId}';
   }
 }
 
-enum RideStatus { preview, pending, approved, rejected, cancelledByDriver, cancelledByRider }
+enum RideStatus { preview, pending, approved, rejected, cancelledByDriver, cancelledByRider, withdrawnByRider }
```

Comment:
what we could also do if we implement the soft delete that we just delete the ride, and remove the withdrawnByRider status. But i'm not sure if that's a good idea, what do you guys think?

Reply:
I think having the status is good. It provides us with information we need. An example: I discovered that if you request a ride, withdraw it, go back, tap on the ride again, and repeat the same thing over and over, you can generate a lot a database entries and it would be good to be able to see that this is because of withdrawn rides

Reply:
What we could do here is actually using a hard delete, but it is a question whether we want/need the information that there formerly was a request.

Reply:
I find the hard delete attractive but its a risk if we let the users delete rides in our database. I think we should keep the status, its better for the cases and its not intuitive that withdrawn rides are inside the hide bool.